exonum/src/encoding/protobuf/mod.rs

```diff
@@ -46,21 +46,20 @@ pub trait ProtobufConvert: Sized {
     fn to_pb(&self) -> Self::ProtoStruct;
 
     /// ProtoStruct -> Struct
-    fn from_pb(pb: Self::ProtoStruct) -> Result<Self, ()>;
+    fn from_pb(pb: Self::ProtoStruct) -> Result<Self, Error>;
```

Comment:
I used failure here since it is common error type for everyone.

Lets say that we have `E1` type for errors for type `A` and `E2` error type for type `B`. If we include struct `B` into `A` than `E2` should be convertible to `E1`(with ? operator), if we include `A` into `B` than `E1` should be convertible to `E2`(with ? operator).

Obvious solution would be to have common error type for everyone to have freedom in composition.